In [66]:
"""
rate= 10000 qp2=qp2*0.35
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[18:48]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[15:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[10:40]
"""
import numpy as np
B=0.0003
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:51.929793638341565, size : 17.730181525735293
time: 1, acc:50.399443682502294, size : 17.22621783088235
time: 2, acc:51.46653231452493, size : 17.460018382352942
time: 3, acc:48.86479938731474, size : 17.34231387867647
time: 4, acc:54.526188794304346, size : 16.922162224264707
time: 5, acc:47.94174643123851, size : 16.87534466911765
time: 6, acc:56.87801417182474, size : 16.640625
time: 7, acc:45.9069336161894, size : 17.29067095588235
time: 8, acc:56.85473049388212, size : 16.82651654411765
time: 9, acc:39.77009268367992, size : 17.466509650735293
time: 10, acc:40.46918083639706, size : 17.473345588235293
time: 11, acc:52.478790283203125, size : 18.013901654411764
time: 12, acc:47.89870767032399, size : 17.756261488970587
time: 13, acc:50.833424399880805, size : 18.998851102941178
time: 14, acc:58.37822240941665, size : 18.169921875
time: 15, acc:45.625790427712836, size : 18.316578584558822
time: 16, acc:50.05683337940889, size : 17.840418198529413
time: 17, acc:62.90923

In [75]:
"""
rate= 10000 qp2=qp2*0.45
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[20:50]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[15:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.0001
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:60.435870114494776, size : 18.928998161764707
time: 1, acc:65.87815845713897, size : 19.102424172794116
time: 2, acc:65.45529646031997, size : 19.38930376838235
time: 3, acc:62.244165644926184, size : 19.240004595588236
time: 4, acc:63.75311402713551, size : 18.776941636029413
time: 5, acc:63.59451798831715, size : 18.818933823529413
time: 6, acc:56.900910770191864, size : 19.17348345588235
time: 7, acc:69.5906358606675, size : 18.631433823529413
time: 8, acc:61.12966817968032, size : 19.310144761029413
time: 9, acc:48.90563067267923, size : 19.40268841911765
time: 10, acc:50.435074637917914, size : 19.216681985294116
time: 11, acc:57.24747882169836, size : 19.882065716911764
time: 12, acc:61.54445479897892, size : 19.946403952205884
time: 13, acc:61.888832204482135, size : 20.886948529411764
time: 14, acc:68.27602105982163, size : 20.09995404411765
time: 15, acc:56.09630977406221, size : 20.25522748161765
time: 16, acc:61.09826985527487, size : 19.60592830882353
time: 17,

In [97]:
"""
rate= 10000 qp2=qp2*0.89
I frame qp=75
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[60:90]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[55:85]
"""
import numpy as np
B=0.00001
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.71007279788746, size : 42.376780790441174
time: 1, acc:96.02361286387725, size : 45.97219669117647
time: 2, acc:95.94868772170123, size : 46.5950137867647
time: 3, acc:93.45225165872013, size : 48.0998965992647
time: 4, acc:95.39385402903838, size : 45.9680606617647
time: 5, acc:95.53351121790269, size : 45.66366038602941
time: 6, acc:95.94988542444565, size : 46.3095703125
time: 7, acc:94.66080104603486, size : 47.13907398897059
time: 8, acc:94.0396084504969, size : 47.280618106617645
time: 9, acc:90.57450855479522, size : 47.523265165441174
time: 10, acc:90.92336822958553, size : 47.384765625
time: 11, acc:93.37265351239373, size : 48.5569278492647
time: 12, acc:94.71547743853401, size : 48.43313419117647
time: 13, acc:96.15319476408116, size : 52.41429227941177
time: 14, acc:95.94931602478027, size : 49.16526884191177
time: 15, acc:90.67880967084099, size : 49.18169806985294
time: 16, acc:94.17810440063477, size : 47.44094669117647
time: 17, acc:92.90662933798397, si

In [110]:
"""
rate= 10000 qp2=qp2*0.68
I frame qp=42
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[28:48]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.00006
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:82.4620667625876, size : 23.851275275735293
time: 1, acc:85.52661783554974, size : 24.972598805147058
time: 2, acc:86.5843464346493, size : 24.890337775735293
time: 3, acc:83.03923606872559, size : 24.70082720588235
time: 4, acc:83.50664307089413, size : 24.928021599264707
time: 5, acc:85.40025318370147, size : 24.13706341911765
time: 6, acc:83.59685505137725, size : 24.644646139705884
time: 7, acc:85.08256743935978, size : 24.935719209558822
time: 8, acc:82.94921482310576, size : 25.093577665441178
time: 9, acc:81.15769554586971, size : 24.85581341911765
time: 10, acc:75.80851386575138, size : 24.458065257352942
time: 11, acc:81.85151324552648, size : 25.342026654411764
time: 12, acc:83.22344667771283, size : 25.80715762867647
time: 13, acc:82.73340393515194, size : 27.13717830882353
time: 14, acc:85.1303409127628, size : 26.284409466911764
time: 15, acc:79.17482151704677, size : 25.732996323529413
time: 16, acc:82.76022181791417, size : 24.741613051470587
time: 17, acc:8

In [114]:
"""
rate= 10000 qp2=qp2*0.7
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[35:65]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[35:65]
"""
import numpy as np
B=0.00005
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:87.61761889738194, size : 25.421702665441178
time: 1, acc:87.40925227894502, size : 25.87339154411765
time: 2, acc:87.03508657567642, size : 26.08352481617647
time: 3, acc:85.56616166058708, size : 25.607364430147058
time: 4, acc:85.69334254545323, size : 25.234317555147058
time: 5, acc:87.47188343721278, size : 25.020220588235293
time: 6, acc:86.08322143554688, size : 25.48110064338235
time: 7, acc:86.15226184620576, size : 25.869198069852942
time: 8, acc:87.14424862581141, size : 25.997242647058822
time: 9, acc:78.3927693086512, size : 25.778722426470587
time: 10, acc:80.16642682692584, size : 25.360236672794116
time: 11, acc:81.02799864376293, size : 26.122817095588236
time: 12, acc:83.03373280693503, size : 26.39407169117647
time: 13, acc:88.50534943973317, size : 28.629136029411764
time: 14, acc:84.8781753988827, size : 27.263614430147058
time: 15, acc:81.28023147583008, size : 26.60707720588235
time: 16, acc:85.00733936534209, size : 25.66831341911765
time: 17, acc:8

In [18]:
"""
rate= 10000 qp2=qp2*0.75
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[35:65]
"""
import numpy as np
B=0.00005001
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:91.39439919415642, size : 26.711167279411764
time: 1, acc:90.3906177071964, size : 27.633559283088236
time: 2, acc:90.1976276846493, size : 27.93606387867647
time: 3, acc:88.73884818133186, size : 27.79917279411765
time: 4, acc:89.3739616169649, size : 26.939280790441178
time: 5, acc:90.49288244808422, size : 27.016888786764707
time: 6, acc:89.3403221579159, size : 27.756606158088236
time: 7, acc:91.24611966750201, size : 28.034122242647058
time: 8, acc:90.99449269911823, size : 28.06985294117647
time: 9, acc:83.02318909588982, size : 28.074448529411764
time: 10, acc:85.62026304357192, size : 27.624253216911764
time: 11, acc:87.63578358818503, size : 28.173828125
time: 12, acc:87.89395164040958, size : 28.72104779411765
time: 13, acc:90.84206468918744, size : 30.478343290441178
time: 14, acc:89.16960884543026, size : 29.693416819852942
time: 15, acc:83.8770165162928, size : 28.726849724264707
time: 16, acc:88.24322083417107, size : 28.059857536764707
time: 17, acc:89.51477

In [22]:
"""
rate= 10000 qp2=qp2*0.78
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00003
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:91.67350039762609, size : 30.290958180147058
time: 1, acc:91.69512356028838, size : 30.790268841911764
time: 2, acc:91.82632390190574, size : 31.15102251838235
time: 3, acc:88.32314996158375, size : 31.854779411764707
time: 4, acc:91.1960573757396, size : 31.284352022058822
time: 5, acc:92.46643010307761, size : 30.27493106617647
time: 6, acc:91.66374767527861, size : 31.468922334558822
time: 7, acc:91.09536058762494, size : 31.25229779411765
time: 8, acc:85.0991922266343, size : 32.12396599264706
time: 9, acc:80.59135605307186, size : 31.948471966911764
time: 10, acc:82.81165571773754, size : 31.552676930147058
time: 11, acc:86.31697262034697, size : 32.48753446691177
time: 12, acc:91.03870111353257, size : 32.262580422794116
time: 13, acc:93.12275718240177, size : 34.774069393382355
time: 14, acc:92.46329139260685, size : 33.3336971507353
time: 15, acc:84.4131161184872, size : 32.17917049632353
time: 16, acc:90.26032616110409, size : 31.632869944852942
time: 17, acc:91.9

In [7]:
"""
rate= 10000 qp2=qp2*0.78
I frame qp=65
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:92.51942354090073, size : 30.112419577205884
time: 1, acc:92.5443873685949, size : 30.987879136029413
time: 2, acc:93.15666591419892, size : 30.749712775735293
time: 3, acc:89.69588560216567, size : 30.836052389705884
time: 4, acc:90.72029730852913, size : 30.291762408088236
time: 5, acc:91.58078642452465, size : 29.950999540441178
time: 6, acc:91.71076662400189, size : 30.489372702205884
time: 7, acc:92.67964924083037, size : 31.037051930147058
time: 8, acc:90.85049068226533, size : 31.166130514705884
time: 9, acc:83.19539743311265, size : 30.899356617647058
time: 10, acc:88.1990068099078, size : 30.412339154411764
time: 11, acc:88.62961039823644, size : 32.021599264705884
time: 12, acc:90.36819233613856, size : 31.528033088235293
time: 13, acc:92.70934497608857, size : 34.28245634191177
time: 14, acc:92.97411582049202, size : 32.603400735294116
time: 15, acc:86.64057675529929, size : 31.876838235294116
time: 16, acc:90.5878319459803, size : 31.489947150735293
time: 17, a

In [11]:
"""
rate= 10000 qp2=qp2*0.77
I frame qp=65
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004001
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:93.45126993515912, size : 30.15676700367647
time: 1, acc:92.74274040670956, size : 30.790843290441178
time: 2, acc:93.48940568811753, size : 31.141486672794116
time: 3, acc:91.26928273369285, size : 31.09897748161765
time: 4, acc:91.08681398279526, size : 30.074046415441178
time: 5, acc:91.61971597110524, size : 30.129940257352942
time: 6, acc:91.7321850271786, size : 30.86965762867647
time: 7, acc:92.64214459587546, size : 31.28659237132353
time: 8, acc:92.41677172043744, size : 31.36960018382353
time: 9, acc:87.68093445721794, size : 31.316808363970587
time: 10, acc:89.4442642436308, size : 30.818589154411764
time: 11, acc:89.21318054199219, size : 31.65625
time: 12, acc:90.94555798698875, size : 31.924689797794116
time: 13, acc:92.87150607389562, size : 34.71139705882353
time: 14, acc:93.07962585898007, size : 33.019186580882355
time: 15, acc:85.52766687729779, size : 32.13045726102941
time: 16, acc:90.37530001472024, size : 31.048426011029413
time: 17, acc:88.756788478

In [15]:
"""
rate= 10000 qp2=qp2*0.82
I frame qp=65
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004002
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:93.48534415749943, size : 32.47162224264706
time: 1, acc:94.49496830210967, size : 34.051585477941174
time: 2, acc:94.59490215077119, size : 34.32180606617647
time: 3, acc:92.81477928161621, size : 34.489372702205884
time: 4, acc:93.38634154375862, size : 33.18514476102941
time: 5, acc:93.96470855264103, size : 33.15406709558823
time: 6, acc:93.87608135447783, size : 33.9108455882353
time: 7, acc:93.9574634327608, size : 34.55819163602941
time: 8, acc:92.23872072556439, size : 34.26441865808823
time: 9, acc:85.22142522475299, size : 35.2448299632353
time: 10, acc:89.10437752218807, size : 35.09886259191177
time: 11, acc:91.18893847746007, size : 34.942727481617645
time: 12, acc:93.24054998510024, size : 35.223460477941174
time: 13, acc:94.42510604858398, size : 38.3381778492647
time: 14, acc:94.40961164586685, size : 36.378561580882355
time: 15, acc:89.14423549876494, size : 35.5778377757353
time: 16, acc:92.0559798969942, size : 35.21042049632353
time: 17, acc:90.04624030

In [61]:
"""
rate= 10000 qp2=qp2*0.65
I frame qp=42
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[37:67]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[32:62]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[25:55]
 """
import numpy as np
B=8.001e-05
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:79.44194008322323, size : 23.060719209558822
time: 1, acc:84.20980397392722, size : 23.768841911764707
time: 2, acc:83.58057246488683, size : 23.703067555147058
time: 3, acc:80.23332147037281, size : 23.485926011029413
time: 4, acc:79.89985802594353, size : 23.695886948529413
time: 5, acc:82.91084065156825, size : 22.956858915441178
time: 6, acc:78.43382218304802, size : 23.443589154411764
time: 7, acc:81.06147261226879, size : 23.69088924632353
time: 8, acc:78.99335131925694, size : 23.889073988970587
time: 9, acc:74.77946561925552, size : 23.533145680147058
time: 10, acc:76.04271664338954, size : 23.242761948529413
time: 11, acc:77.36524133121266, size : 24.262522977941178
time: 12, acc:79.38839688020595, size : 24.647690716911764
time: 13, acc:81.81557935826919, size : 26.437270220588236
time: 14, acc:81.40884006724639, size : 25.072150735294116
time: 15, acc:76.10482608570771, size : 24.534869025735293
time: 16, acc:77.54996524137609, size : 23.620806525735293
time: 17

Version1 

In [4]:
"""
qp2=qp2*0.45
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[18:48]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[15:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[10:40]
"""
import numpy as np
B=0.000301
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:45.581652136409986, size : 17.621323529411764
time: 1, acc:39.35474507948931, size : 16.461799172794116
time: 2, acc:38.50757374482996, size : 17.24856387867647
time: 3, acc:52.590510424445654, size : 17.476677389705884
time: 4, acc:60.32663794124828, size : 17.092486213235293
time: 5, acc:48.30222410314224, size : 16.912798713235293
time: 6, acc:30.84115982055664, size : 16.319623161764707
time: 7, acc:39.23147425932043, size : 17.374080882352942
time: 8, acc:33.808438918169806, size : 16.800723805147058
time: 9, acc:30.501864938174975, size : 17.15963924632353
time: 10, acc:33.715388354133154, size : 17.90234375
time: 11, acc:43.563169591567096, size : 17.821116727941178
time: 12, acc:41.77023943732767, size : 17.9716796875
time: 13, acc:46.94510908687816, size : 19.065487132352942
time: 14, acc:50.91670821694767, size : 17.71547564338235
time: 15, acc:54.465554742252124, size : 18.067784926470587
time: 16, acc:42.50083250157974, size : 17.484777113970587
time: 17, acc:5

In [5]:
"""
 qp2=qp2*0.6
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[20:50]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[15:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.000101
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:52.67674782696892, size : 19.337316176470587
time: 1, acc:49.16244114146513, size : 19.216509650735293
time: 2, acc:43.46421466154211, size : 18.874425551470587
time: 3, acc:45.90711032643038, size : 19.025505514705884
time: 4, acc:47.233292635749365, size : 18.71627987132353
time: 5, acc:44.37389373779297, size : 18.540441176470587
time: 6, acc:45.40999917423024, size : 18.9287109375
time: 7, acc:45.26865342084099, size : 19.174747242647058
time: 8, acc:52.08884968477137, size : 19.4072265625
time: 9, acc:47.59980650509105, size : 19.477711397058822
time: 10, acc:59.072337431066174, size : 19.43600643382353
time: 11, acc:51.140333624447095, size : 19.754940257352942
time: 12, acc:57.45913000667797, size : 20.35696231617647
time: 13, acc:61.93196352790384, size : 21.311810661764707
time: 14, acc:61.30049649406882, size : 20.477309283088236
time: 15, acc:62.278357674093805, size : 20.400620404411764
time: 16, acc:57.18621366164263, size : 19.932329963235293
time: 17, acc:64

In [6]:
"""
 qp2=qp2*0.92
I frame qp=75
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[60:90]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[55:85]
"""
import numpy as np
B=0.0000101
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.67354987649357, size : 42.1891659007353
time: 1, acc:95.61059054206399, size : 45.139476102941174
time: 2, acc:95.43610460617963, size : 46.02809053308823
time: 3, acc:93.97250904756434, size : 45.960133272058826
time: 4, acc:95.64735468696145, size : 45.547506893382355
time: 5, acc:95.59256609748391, size : 46.62574678308823
time: 6, acc:95.59827692368451, size : 45.43313419117647
time: 7, acc:95.65210622899673, size : 49.14056755514706
time: 8, acc:94.35281192555146, size : 46.20944393382353
time: 9, acc:92.24844539866729, size : 47.28705193014706
time: 10, acc:92.94040062848259, size : 47.14918428308823
time: 11, acc:94.84972953796387, size : 48.41348805147059
time: 12, acc:94.87932990579044, size : 47.3659237132353
time: 13, acc:96.06722943923053, size : 52.82961856617647
time: 14, acc:95.58942177716423, size : 48.15194163602941
time: 15, acc:91.81349698235007, size : 49.08295036764706
time: 16, acc:94.37874064725987, size : 47.048196231617645
time: 17, acc:92.28728

In [17]:
"""
 qp2=qp2*0.72
I frame qp=42
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[28:48]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[15:45]
"""
import numpy as np
B=0.0000601
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:77.01004252714269, size : 22.79917279411765
time: 1, acc:83.53715503917022, size : 24.438189338235293
time: 2, acc:84.43731700672822, size : 24.875287224264707
time: 3, acc:78.43199056737564, size : 24.754710477941178
time: 4, acc:81.10588859109318, size : 24.158203125
time: 5, acc:82.68203735351562, size : 24.121783088235293
time: 6, acc:81.94394111633301, size : 24.67043887867647
time: 7, acc:79.60873491623822, size : 23.890337775735293
time: 8, acc:77.70796663620892, size : 24.956801470588236
time: 9, acc:77.79432465048397, size : 25.298081341911764
time: 10, acc:70.06208475898293, size : 23.795323988970587
time: 11, acc:76.58191007726333, size : 25.55325137867647
time: 12, acc:75.12798028833726, size : 25.009765625
time: 13, acc:83.63238222458783, size : 27.012522977941178
time: 14, acc:80.8785410488353, size : 25.722541360294116
time: 15, acc:73.91168930951287, size : 25.353343290441178
time: 16, acc:73.0859027189367, size : 23.888901654411764
time: 17, acc:88.8891612

In [22]:
"""
 qp2=qp2*0.8
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[35:65]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[35:65]
"""
import numpy as np
B=0.00005002
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:86.88939038444968, size : 25.589613970588236
time: 1, acc:87.29844654307647, size : 25.555549172794116
time: 2, acc:86.99156536775477, size : 25.910271139705884
time: 3, acc:82.98089083503274, size : 25.760569852941178
time: 4, acc:83.36586952209473, size : 25.41555606617647
time: 5, acc:85.30296998865464, size : 25.403377757352942
time: 6, acc:84.43763396319221, size : 25.66946231617647
time: 7, acc:86.37344416450051, size : 26.106272977941178
time: 8, acc:82.75623321533203, size : 25.949793198529413
time: 9, acc:79.47368060841279, size : 26.02194393382353
time: 10, acc:79.36700372134938, size : 25.701918658088236
time: 11, acc:78.78132146947524, size : 26.564682904411764
time: 12, acc:79.38461023218491, size : 26.9248046875
time: 13, acc:85.0750867058249, size : 28.478285845588236
time: 14, acc:83.17838276133818, size : 27.70461856617647
time: 15, acc:82.07185969633214, size : 26.93997012867647
time: 16, acc:72.96713099760167, size : 26.480238970588236
time: 17, acc:77.2

In [33]:
"""
 qp2=qp2*0.78
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[40:70]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[35:65]
"""
import numpy as np
B=0.00005003
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:87.83818974214442, size : 25.7451171875
time: 1, acc:89.52268712660846, size : 27.358455882352942
time: 2, acc:88.9169552746941, size : 27.846507352941178
time: 3, acc:85.25604079751407, size : 27.736155790441178
time: 4, acc:88.09092465568992, size : 27.096277573529413
time: 5, acc:88.5465145111084, size : 27.072552849264707
time: 6, acc:88.214952805463, size : 27.50258501838235
time: 7, acc:85.66283618702607, size : 26.524299172794116
time: 8, acc:83.17291876849006, size : 27.956629136029413
time: 9, acc:78.19573178010829, size : 26.891773897058822
time: 10, acc:78.7910545573515, size : 26.469956341911764
time: 11, acc:85.37596814772662, size : 28.38803998161765
time: 12, acc:85.4795119341682, size : 27.928481158088236
time: 13, acc:91.63991142721737, size : 30.320255055147058
time: 14, acc:87.40760017843807, size : 28.674574908088236
time: 15, acc:79.04768551097197, size : 28.134535845588236
time: 16, acc:79.97620245989631, size : 26.673023897058822
time: 17, acc:79.297

In [37]:
"""
 qp2=qp2*0.82
I frame qp=55
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.0000301
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:90.18457076128792, size : 27.720760569852942
time: 1, acc:91.33940023534439, size : 30.413315716911764
time: 2, acc:91.24140458948472, size : 31.091107536764707
time: 3, acc:88.45392395468319, size : 30.979894301470587
time: 4, acc:92.05321985132554, size : 30.285845588235293
time: 5, acc:92.53997522241929, size : 30.28176700367647
time: 6, acc:92.72355472340303, size : 30.920036764705884
time: 7, acc:89.74170684814453, size : 31.519761029411764
time: 8, acc:86.62542174844181, size : 31.478745404411764
time: 9, acc:81.8828779108384, size : 31.90785845588235
time: 10, acc:84.1143383699305, size : 31.566578584558822
time: 11, acc:87.03100541058708, size : 32.31939338235294
time: 12, acc:88.16691566916073, size : 31.329159007352942
time: 13, acc:93.30934917225557, size : 34.095818014705884
time: 14, acc:90.860055474674, size : 31.935546875
time: 15, acc:86.15905256832347, size : 33.02493106617647
time: 16, acc:87.34641355626724, size : 29.657284007352942
time: 17, acc:92.3855

In [7]:

"""
 qp2=qp2*0.85
I frame qp=65
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004003
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:93.0482780232149, size : 30.081112132352942
time: 1, acc:92.2602232764749, size : 29.993910845588236
time: 2, acc:91.80038676542394, size : 30.7880859375
time: 3, acc:90.71115886463838, size : 30.640567555147058
time: 4, acc:92.06273135016946, size : 30.11391314338235
time: 5, acc:91.416737612556, size : 29.925666360294116
time: 6, acc:91.61307671490837, size : 30.3095703125
time: 7, acc:91.64643568151138, size : 30.652171415441178
time: 8, acc:91.890708137961, size : 30.990062040441178
time: 9, acc:87.0874096365536, size : 31.051815257352942
time: 10, acc:90.31393948723289, size : 30.789981617647058
time: 11, acc:88.17491811864517, size : 31.369427849264707
time: 12, acc:89.98852617600384, size : 32.107536764705884
time: 13, acc:91.92033375010772, size : 34.13373161764706
time: 14, acc:88.84342137505027, size : 32.457203584558826
time: 15, acc:84.41488602582146, size : 32.2431640625
time: 16, acc:85.79387103810029, size : 31.608226102941178
time: 17, acc:83.38299358592315

In [10]:

"""
 qp2=qp2*0.87
I frame qp=65
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[55:85]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004004
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:93.98919273825253, size : 32.06341911764706
time: 1, acc:93.8510165495031, size : 33.673943014705884
time: 2, acc:94.59935917573817, size : 34.11965762867647
time: 3, acc:92.1545757966883, size : 33.88229549632353
time: 4, acc:93.80166670855354, size : 32.88355928308823
time: 5, acc:93.51018176359288, size : 32.57835477941177
time: 6, acc:93.84126943700454, size : 32.98977481617647
time: 7, acc:93.81258908440086, size : 33.95875459558823
time: 8, acc:92.35059794257668, size : 34.26246553308823
time: 9, acc:88.92122436972225, size : 34.805491727941174
time: 10, acc:90.79781981075512, size : 34.505514705882355
time: 11, acc:90.79052139731014, size : 34.536305147058826
time: 12, acc:93.16539764404297, size : 35.50436580882353
time: 13, acc:94.19131559484146, size : 37.438648897058826
time: 14, acc:91.71417180229636, size : 35.4856387867647
time: 15, acc:86.96479797363281, size : 35.78952205882353
time: 16, acc:91.59577594083898, size : 34.57864200367647
time: 17, acc:89.18955

In [11]:
"""
qp2=qp2*0.7
I frame qp=42
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[37:67]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[32:62]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[25:55]
 """
import numpy as np
B=8.002e-05
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:75.41691836188821, size : 22.287798713235293
time: 1, acc:82.54273078020881, size : 23.58628216911765
time: 2, acc:82.32644025017234, size : 24.011431525735293
time: 3, acc:76.89958179698272, size : 23.89211856617647
time: 4, acc:79.34622764587402, size : 23.31870404411765
time: 5, acc:80.34723225761863, size : 23.302906709558822
time: 6, acc:77.02279090881348, size : 23.707892922794116
time: 7, acc:72.46071591096766, size : 22.873736213235293
time: 8, acc:76.06706058277803, size : 24.066119025735293
time: 9, acc:63.800556519452265, size : 23.148609834558822
time: 10, acc:64.45753995110007, size : 22.790441176470587
time: 11, acc:73.77248932333553, size : 24.46518841911765
time: 12, acc:69.67147939345416, size : 24.1513671875
time: 13, acc:82.99495472627528, size : 26.0869140625
time: 14, acc:78.3054464003619, size : 24.799574908088236
time: 15, acc:71.58410128425149, size : 24.251780790441178
time: 16, acc:69.82760429382324, size : 23.157973345588236
time: 17, acc:74.6279

In [18]:
"""
qp2=qp2*0.9
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1e-06
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.4579886268167, size : 47.765797334558826
time: 1, acc:95.18686743343578, size : 48.38579963235294
time: 2, acc:95.18161661484662, size : 49.5618106617647
time: 3, acc:93.2120351230397, size : 49.95387178308823
time: 4, acc:95.28754458707921, size : 48.64280790441177
time: 5, acc:95.52760685191436, size : 49.11667049632353
time: 6, acc:95.34409466911765, size : 49.637580422794116
time: 7, acc:94.5260721094468, size : 50.58358226102941
time: 8, acc:92.79739716473748, size : 49.911937040441174
time: 9, acc:91.13321865306182, size : 50.3557559742647
time: 10, acc:91.72962132622214, size : 49.84914981617647
time: 11, acc:93.50136588601505, size : 51.025735294117645
time: 12, acc:94.79978785795323, size : 52.11764705882353
time: 13, acc:95.2552739311667, size : 55.6301700367647
time: 14, acc:95.62914231244255, size : 53.45306755514706
time: 15, acc:91.2916996899773, size : 52.44525505514706
time: 16, acc:94.55764433916877, size : 49.992532169117645
time: 17, acc:95.3623911913

In [19]:
"""
qp2=qp2*0.95
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1.001e-06
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:96.71501271864948, size : 63.45806525735294
time: 1, acc:96.44965284010944, size : 73.3388671875
time: 2, acc:96.67399630827062, size : 75.41923253676471
time: 3, acc:95.1113785014433, size : 76.45318244485294
time: 4, acc:96.32372295155244, size : 74.69134880514706
time: 5, acc:96.45502707537483, size : 75.47782628676471
time: 6, acc:96.26767775591682, size : 75.71536075367646
time: 7, acc:95.9323995253619, size : 77.59771369485294
time: 8, acc:95.0894411872415, size : 76.06497012867646
time: 9, acc:93.47509215859806, size : 77.3404181985294
time: 10, acc:94.47050094604492, size : 76.89447380514706
time: 11, acc:95.34398247213925, size : 78.39234834558823
time: 12, acc:95.92861568226533, size : 79.32525275735294
time: 13, acc:96.53661110821892, size : 84.40613511029412
time: 14, acc:96.85364891501034, size : 80.693359375
time: 15, acc:92.68010924844181, size : 80.07261029411765
time: 16, acc:95.81091263714958, size : 76.96300551470588
time: 17, acc:96.24871927149155, size

In [20]:
"""
qp2=qp2*0.97
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1.001e-07
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:96.9140277189367, size : 80.59392233455883
time: 1, acc:96.97448786567239, size : 99.46225873161765
time: 2, acc:96.91097035127528, size : 102.30922564338235
time: 3, acc:95.49166454988368, size : 104.47960707720588
time: 4, acc:96.70875773710362, size : 102.42549402573529
time: 5, acc:96.79419012630687, size : 103.21811810661765
time: 6, acc:96.69226197635426, size : 103.46237362132354
time: 7, acc:96.2636246400721, size : 104.43037683823529
time: 8, acc:95.73135656469009, size : 103.95088465073529
time: 9, acc:93.98321263930377, size : 104.07548253676471
time: 10, acc:94.94448830099667, size : 103.85966222426471
time: 11, acc:95.9485138163847, size : 105.08432904411765
time: 12, acc:96.19949565214269, size : 107.26694623161765
time: 13, acc:96.87580781824448, size : 113.35340073529412
time: 14, acc:97.24146618562587, size : 109.77849264705883
time: 15, acc:93.3411205516142, size : 107.32915900735294
time: 16, acc:96.06453671174891, size : 103.42026654411765
time: 17, acc

In [21]:
"""
qp2=qp2*0.99
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1.002e-07
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:97.012399224674, size : 105.1019646139706
time: 1, acc:97.19353283152861, size : 130.45209099264707
time: 2, acc:97.1570800332462, size : 136.56382123161765
time: 3, acc:95.97881261040183, size : 142.58392693014707
time: 4, acc:96.78594364839442, size : 135.68629365808823
time: 5, acc:97.00940076042625, size : 146.0908203125
time: 6, acc:96.96092325098374, size : 141.5232651654412
time: 7, acc:96.59281337962432, size : 142.9342830882353
time: 8, acc:95.96320601070629, size : 139.0439453125
time: 9, acc:94.20246516957002, size : 141.15975413602942
time: 10, acc:95.300853953642, size : 143.0568129595588
time: 11, acc:96.01819374982048, size : 145.30261948529412
time: 12, acc:96.3727081523222, size : 143.46461397058823
time: 13, acc:97.07415244158577, size : 156.75735294117646
time: 14, acc:97.58783789242015, size : 148.66819852941177
time: 15, acc:93.69089463177849, size : 143.13803998161765
time: 16, acc:96.35031924528234, size : 145.8829848345588
time: 17, acc:96.525943980

In [24]:
"""
qp2=qp2*0.0.35
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[0:30]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[0:30]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[0:30]
 """
import numpy as np
B=0.001
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:35.645069795496326, size : 13.562270220588236
time: 1, acc:39.892931545481964, size : 12.683306525735293
time: 2, acc:38.94753736608169, size : 12.938304227941176
time: 3, acc:38.36746496312759, size : 12.987132352941176
time: 4, acc:35.436201095581055, size : 12.557100183823529
time: 5, acc:36.048316955566406, size : 12.173943014705882
time: 6, acc:36.756554771872125, size : 12.314108455882353
time: 7, acc:37.485913669361786, size : 12.500517003676471
time: 8, acc:34.67661352718578, size : 12.878159466911764
time: 9, acc:28.430938720703125, size : 13.254997702205882
time: 10, acc:35.71343982920927, size : 13.367647058823529
time: 11, acc:32.92617517359116, size : 13.171070772058824
time: 12, acc:37.77766508214614, size : 13.375804227941176
time: 13, acc:38.5896037606632, size : 13.545840992647058
time: 14, acc:48.11910741469439, size : 13.211856617647058
time: 15, acc:44.28576581618365, size : 13.363396139705882
time: 16, acc:41.68633573195513, size : 13.510282628676471
t

Version2 
B=0.001     39.92                ACC:    37.49         
B=1.002e-07        96.00    ACC:      89.60
B=1.001e-07        95.77    ACC:    89.54
B=1.001e-06        95.37    ACC:   89.35
B=8.002e-05         73.76   ACC:    73.06
B=0.00004004     90.51   ACC:     86.75
B=0.00004003     87.31   ACC:     84.48
B=0.0000301       88.81   ACC:     84.85
B=0.0000601       78.33    ACC:    77.18
B=0.0000101      94.21    ACC:     88.86
B=0.000101        55.29    ACC:      55.65
B=0.000301        46.82    ACC:     47.18

Version2  gaibianle yuyifenge chuanbo

In [2]:
"""
qp2=qp2*0.95
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1.002e-06
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.19527940189137, size : 64.28469669117646
time: 1, acc:93.98024222430061, size : 49.43480009191177
time: 2, acc:93.04553200216854, size : 50.51470588235294
time: 3, acc:91.23017647687126, size : 52.02809053308823
time: 4, acc:92.63526972602395, size : 50.73753446691177
time: 5, acc:92.43841451757095, size : 51.34719669117647
time: 6, acc:92.11871483746697, size : 51.466624540441174
time: 7, acc:91.5535898769603, size : 52.3889590992647
time: 8, acc:89.83955663793228, size : 51.82864200367647
time: 9, acc:84.8495034610524, size : 52.66205193014706
time: 10, acc:86.27137857324936, size : 51.995289522058826
time: 11, acc:89.46832208072438, size : 52.759880514705884
time: 12, acc:90.32050020554486, size : 53.180836397058826
time: 13, acc:91.46240178276511, size : 56.523265165441174
time: 14, acc:93.63366014817181, size : 54.4108455882353
time: 15, acc:83.5234052994672, size : 53.62281709558823
time: 16, acc:89.02615659377155, size : 52.825827205882355
time: 17, acc:91.912499

In [4]:
"""
qp2=qp2*0.7
I frame qp=42
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[37:67]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[32:62]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[25:55]
 """
import numpy as np
B=8.003e-05
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:73.59740032869227, size : 22.2880859375
time: 1, acc:75.0320743112003, size : 15.956801470588236
time: 2, acc:80.8469435747932, size : 16.162224264705884
time: 3, acc:67.20321879667394, size : 15.181123621323529
time: 4, acc:65.50285675946404, size : 15.598977481617647
time: 5, acc:57.93556606068331, size : 14.645967371323529
time: 6, acc:68.73932445750518, size : 15.924057904411764
time: 7, acc:66.70329991508933, size : 15.234432444852942
time: 8, acc:69.5772928350112, size : 16.101447610294116
time: 9, acc:71.2127264808206, size : 15.437442555147058
time: 10, acc:63.66206617916332, size : 15.339326746323529
time: 11, acc:61.2158691181856, size : 16.30141314338235
time: 12, acc:70.48137889188878, size : 15.970128676470589
time: 13, acc:72.96170907862046, size : 16.700942095588236
time: 14, acc:79.28835363949047, size : 16.418772977941178
time: 15, acc:68.38870048522949, size : 16.19789751838235
time: 16, acc:63.17114269032198, size : 16.297162224264707
time: 17, acc:83.25

In [8]:

"""
 qp2=qp2*0.85
I frame qp=65
             self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[50:80]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[45:75]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[45:75]
"""
import numpy as np
B=0.00004005
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:92.62199401855469, size : 30.079331341911764
time: 1, acc:91.52409890118767, size : 20.35696231617647
time: 2, acc:91.05664141037884, size : 20.39108455882353
time: 3, acc:88.63936031565947, size : 20.607996323529413
time: 4, acc:91.20596717385685, size : 19.816693474264707
time: 5, acc:88.14005010268268, size : 20.331227022058822
time: 6, acc:90.63750996309169, size : 20.628676470588236
time: 7, acc:89.97183126561782, size : 20.955767463235293
time: 8, acc:86.76801849814022, size : 20.952378216911764
time: 9, acc:79.4878174276913, size : 21.051930147058822
time: 10, acc:83.50614940418916, size : 20.74253216911765
time: 11, acc:83.91349455889534, size : 21.421645220588236
time: 12, acc:86.17379805620979, size : 21.51418887867647
time: 13, acc:88.33166851716884, size : 23.124425551470587
time: 14, acc:91.03111940271714, size : 21.929400275735293
time: 15, acc:74.80205367593204, size : 21.021656709558822
time: 16, acc:84.29185362423168, size : 21.18215762867647
time: 17, acc

In [9]:
"""
qp2=qp2*0.45
I frame qp=35
            self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[18:48]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[15:45]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[10:40]
"""
import numpy as np
B=0.000302
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:53.387871910544, size : 17.621036305147058
time: 1, acc:38.75920632306267, size : 11.128791360294118
time: 2, acc:30.247937931733972, size : 11.544462316176471
time: 3, acc:29.571558447445142, size : 11.874942555147058
time: 4, acc:34.080157560460705, size : 10.625746783088236
time: 5, acc:27.454516466926126, size : 11.272403492647058
time: 6, acc:26.878847795374252, size : 11.553481158088236
time: 7, acc:35.01193944145651, size : 11.115579044117647
time: 8, acc:30.47779868630802, size : 11.766027113970589
time: 9, acc:42.84757445840275, size : 11.420783547794118
time: 10, acc:39.745283126831055, size : 11.511201746323529
time: 11, acc:31.581800124224493, size : 12.147173713235293
time: 12, acc:47.710295284495636, size : 11.868738511029411
time: 13, acc:48.94409460179946, size : 12.232306985294118
time: 14, acc:51.375520930570715, size : 12.061868106617647
time: 15, acc:47.89542590870577, size : 11.703814338235293
time: 16, acc:53.852086908677045, size : 11.725126378676471

In [10]:
"""
qp2=qp2*0.97
I frame qp=85
self.tables1Y = torch.load('./pth/jpeg_qtables.pth')[70:100]
            self.tables1U = torch.load('./pth/jpeg_qtables.pth')[65:95]
            self.tables1V = torch.load('./pth/jpeg_qtables.pth')[60:90]
 """
import numpy as np
B=1.004e-07
fps=3.0
filename="fps={0}/test-B={1}-fps={2}.log".format(int(fps),B,fps)
file = open(filename, 'r')
line=0
loss_data=[]
second_data=[]
V_data=[]
for eachline in file:
#   if line>34*3:break
  if eachline.find("Loss") >= 0 and line>0:
      loss_data.append(eachline)
  if eachline.find("compressed size") >= 0 and line>0:
      V_data.append(eachline)
  if eachline.find("Average Acc") >=0 :
    second_data.append(eachline)
  line+=1
file.close()
# every second average
second_acc=[]
second_size=[]
for info in second_data:
    info=info.split(" ")
    second_acc.append(info[2])
    second_acc.append(info[5])
    print("time: {0}, acc:{1}, size : {2}".format(info[27].strip('\n'),info[8],info[16]))
loss=[]
acc=[]
V=[]
index=0
# 599 average
for info in loss_data:
    info=info.split(" ")[4].split(",")
    acc.append(float(info[1].split(":")[1]))
    index+=1
average_acc=np.mean(np.abs(np.array(acc)))
print("Average Acc:{0}".format(average_acc))
index=0
for info in V_data:
    info=info.split(",")
    V.append(float(info[-2].split(":")[1]))
    index+=1
average_V=np.mean(np.abs(np.array(V)))
print("Average size:{0}".format(average_V))

time: 0, acc:95.23060181561638, size : 81.44927619485294
time: 1, acc:94.24177338095272, size : 67.20571001838235
time: 2, acc:93.26001335592831, size : 68.44370404411765
time: 3, acc:91.43452644348145, size : 70.56887637867646
time: 4, acc:92.60758231667911, size : 69.08507582720588
time: 5, acc:92.49625205993652, size : 70.16946231617646
time: 6, acc:92.4553394317627, size : 70.03308823529412
time: 7, acc:91.63891006918514, size : 70.36029411764706
time: 8, acc:90.55191488826976, size : 70.60546875
time: 9, acc:85.15545901130227, size : 70.84018841911765
time: 10, acc:86.24109099893009, size : 69.79176240808823
time: 11, acc:89.55931663513184, size : 70.65337775735294
time: 12, acc:90.51450000089757, size : 72.10317095588235
time: 13, acc:91.53052217820111, size : 75.63625919117646
time: 14, acc:93.93195264479694, size : 73.712890625
time: 15, acc:84.16560397428624, size : 71.93353630514706
time: 16, acc:89.30460424984203, size : 70.57605698529412
time: 17, acc:92.05543854657341, siz